trulens_eval/trulens_eval/streamlit.py

```diff
@@ -0,0 +1,242 @@
+                    delta_color=(
+                        "normal" if cat.compare(
+                            mean, CATEGORY.PASS[cat.direction].threshold
+                        ) else "inverse"
+                    ),
+                )
+
+        st.markdown("""---""")
+
+
+@st.experimental_fragment(run_every=2)
+def trulens_feedback(record: Record):
+    """
+    Render clickable feedback pills for a given record.
+
+    Args:
+
+        record (Record): A trulens record.
+
+    !!! example
+
+        ```python
+        from trulens_eval import streamlit as trulens_st
+
+        with tru_llm as recording:
+            response = llm.invoke(input_text)
+
+        record, response = recording.get()
+
+        trulens_st.trulens_feedback(record=record)
+        ```
+    """
+    feedback_cols = []
+    feedbacks = {}
+    icons = []
+    for feedback, feedback_result in record.wait_for_feedback_results().items():
+        call_data = {
+            'feedback_definition': feedback,
+            'feedback_name': feedback.name,
+            'result': feedback_result.result
+        }
+        feedback_cols.append(call_data['feedback_name'])
+        feedbacks[call_data['feedback_name']] = FeedbackDisplay(
+            score=call_data['result'],
+            calls=[],
+            icon=display.get_icon(fdef=feedback, result=feedback_result.result)
+        )
+        icons.append(feedbacks[call_data['feedback_name']].icon)
+
+    st.write('**Feedback functions**')
+    selected_feedback = pills(
+        "Feedback functions",
+        feedback_cols,
+        index=None,
+        format_func=lambda fcol: f"{fcol} {feedbacks[fcol].score:.4f}",
+        label_visibility=
+        "collapsed",  # Hiding because we can't format the label here.
+        icons=icons,
+        key=
+        f"{call_data['feedback_name']}_{len(feedbacks)}"  # Important! Otherwise streamlit sometimes lazily skips update even with st.experimental_fragment
+    )
+
+    if selected_feedback is not None:
+        st.dataframe(
+            display.get_feedback_result(
+                record, feedback_name=selected_feedback
+            ),
+            use_container_width=True,
+            hide_index=True
+        )
+
+
+def trulens_trace(record: Record):
+    """
+    Display the trace view for a record.
+
+    Args:
+
+        record (Record): A trulens record.
+
+    !!! example
+
+        ```python
+        from trulens_eval import streamlit as trulens_st
+
+        with tru_llm as recording:
+            response = llm.invoke(input_text)
+            
+        record, response = recording.get()
+
+        trulens_st.trulens_trace(record=record)
+        ```
+    """
+
+    tru = Tru()
+
+    app = tru.get_app(app_id=record.app_id)
+    records, feedback = tru.get_records_and_feedback()
+    record_json = json.loads(
+        records.loc[records['record_id'] == record.record_id]
+        ['record_json'].values[0]
+    )
```

Comment:
@sfc-gh-jreini curious - could this not just be `json_str_of_obj(record)`? Looking at https://github.com/truera/trulens/blob/b95d5ed954a5e4566fa505fa883eb23d94e97a7a/trulens_eval/trulens_eval/database/orm.py#L242

Reply:
Thanks - this works. Way cleaner :)